Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -158,9 +158,9 @@ namespace Terrain
         {
             m_parentScene->GetViewTagBitRegistry().ReleaseTag(m_meshMovedFlag);
         }
+        RemoveRayTracedMeshes();
         m_candidateSectors.clear();
         m_sectorsThatNeedSrgCompiled.clear();
-        RemoveRayTracedMeshes();
         m_sectorLods.clear();
         m_xyPositions.clear();
         m_cachedDrawData.clear();
```

Comment:
There are several repeats of this clear logic throughout this file.

Perhaps we can make it safer by having this code in a dedicated function such as `ClearTerrainSectors`?

Reply:
Actually this code is inside of the `TerrainMeshManager::Reset` function.

Can this function be called below then instead of the explicit calls to `RemoveRayTracedMeshes()` followed by `m_sectorLods.clear()`, etc...?

Reply:
I went with the first suggestion and consolidated the sector clears into a single method. Reset() clears more than that, so I don't feel comfortable switching everything to use that. It's _probably_ ok, but this is more trivially proveable to be correct.